src/l500/l500-options.cpp

```diff
@@ -63,28 +100,81 @@ namespace librealsense
         auto max_value = float(*(reinterpret_cast<int32_t*>(max.data())));
         auto min_value = float(*(reinterpret_cast<int32_t*>(min.data())));
 
+        hwmon_response response;
+        auto res = query_default( &response );
+
+        if (response == hwm_IllegalHwState)
+        {
+            _is_read_only = true;
+            res = -1;
+        }
+
         _range = option_range{ min_value,
-            max_value,
-            float(*(reinterpret_cast<int32_t*>(step.data()))),
-            def };
+                               max_value,
+                               float( *( reinterpret_cast< int32_t * >( step.data() ) ) ),
+                               res };
+    }
+
+    void l500_hw_options::update_default( float def )
+    {
+        _range.def = def;
+    }
+
+    void l500_hw_options::set_read_only( bool read_only )
+    {
+        _is_read_only = read_only;
+    }
+
+    void l500_hw_options::set_manually( bool set ) 
+    {
+        _was_set_manually = set;
     }
 
     void l500_hw_options::enable_recording(std::function<void(const option&)> recording_action)
     {}
 
-    float l500_hw_options::query(int mode) const
+    float l500_hw_options::get_default( int mode, hwmon_response * response ) const
     {
-        auto res = _hw_monitor->send(command{ AMCGET, _type, get_current, mode });
-
-        if (res.size() < sizeof(int32_t))
+        auto res = _hw_monitor->send( command{ AMCGET, _type, l500_command::get_default, mode },
+                                      response );
```

Comment:
`send` should never throw -- always pass a response pointer to it

Reply:
done